### PR TITLE
feat(gitlabci-include): add support for 'include:local'

### PR DESCRIPTION
**Pull request by j0nm1**
Adds support for `include:local` within `gitlab-ci.yml` files (https://docs.gitlab.com/ee/ci/yaml/#includelocal). Currently Renovate ignores these files.

### CODE REVIEW

File: lib/manager/gitlabci-include/extract.ts [hunk 1]
```diff
@@ -3,16 +3,15 @@ import yaml from 'js-yaml';
...
+import { readLocalFile } from '../../util/gitfs';
 import { ExtractConfig, PackageDependency, PackageFile } from '../common';
+import * as gitlabci from '../gitlabci/extract';
 
-function extractDepFromInclude(includeObj: {
+function extractDepFromIncludeFile(includeObj: {
   file: any;
   project: string;
   ref: string;
 }): PackageDependency | null {
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:14]:
```suggestion
}): PackageDependency {
```

File: lib/manager/gitlabci-include/extract.ts [hunk 2]
```diff
@@ -26,22 +25,37 @@ function extractDepFromInclude(includeObj: {
...
     const doc = yaml.safeLoad(content, { json: true });
     if (doc?.include && is.array(doc.include)) {
       for (const includeObj of doc.include) {
-        const dep = extractDepFromInclude(includeObj);
-        if (dep) {
-          if (config.endpoint) {
-            dep.registryUrls = [config.endpoint.replace(/\/api\/v4\/?/, '')];
+        if (includeObj.file && includeObj.project) {
+          const dep = extractDepFromIncludeFile(includeObj);
+          if (dep) {
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:48]:
```suggestion
```

Is now never null

File: lib/manager/gitlabci-include/extract.ts [hunk 3]
```diff
@@ -26,22 +25,37 @@ function extractDepFromInclude(includeObj: {
...
+            if (config.endpoint) {
+              dep.registryUrls = [config.endpoint.replace(/\/api\/v4\/?/, '')];
+            }
+            deps.push(dep);
+          }
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:53]:
```suggestion
```

File: lib/manager/gitlabci-include/extract.ts [hunk 4]
```diff
@@ -26,22 +25,37 @@ function extractDepFromInclude(includeObj: {
   return dep;
 }
 
-export function extractPackageFile(
+async function extractDepsFromIncludeLocal(includeObj: {
+  local: string;
+}): Promise<PackageDependency[] | null> {
+  const content = await readLocalFile(includeObj.local, 'utf8');
+  const deps: PackageDependency[] = gitlabci.extractPackageFile(content).deps;
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:32]:
```suggestion
  const deps = gitlabci.extractPackageFile(content)?.deps;
```

Result can be null

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/manager/gitlabci-include/extract.ts [hunk 5]
```diff
@@ -26,22 +25,35 @@ function extractDepFromInclude(includeObj: {
...
-        if (dep) {
+        if (includeObj.file && includeObj.project) {
+          const dep = extractDepFromIncludeFile(includeObj);
           if (config.endpoint) {
             dep.registryUrls = [config.endpoint.replace(/\/api\/v4\/?/, '')];
           }
           deps.push(dep);
+        } else if (includeObj.local) {
+          const includedDeps = await extractDepsFromIncludeLocal(includeObj);
+          for (const includedDep of includedDeps) {
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:54]: `includedDeps` needs to be checked because it can be `null` which isn't iterable.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/manager/gitlabci-include/extract.ts (hunk 5 above)
Comment (j0nm1) [reply lib/manager/gitlabci-include/extract.ts]: added includedDeps null check

**Review by viceice — APPROVED** [review]

**Pull request merged**